src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -934,40 +1025,71 @@ private void clearSourceFromLayer(int destId) throws Exception {
-			firstLayerDTO.setDimension(layerDimension);
+			LayerDTO firstLayerDTO = prepareLayerDTO(layerIdx, getSourceIndex, 1, 0, layerDimension, layerNode);
 			layerDTOList.add(firstLayerDTO);
-			LayerDTO secondLayerDTO = new LayerDTO(layerId + 1, getSourceIndex, 0, 1);
-			secondLayerDTO.setDimension(layerDimension);
+			LayerDTO secondLayerDTO = prepareLayerDTO(layerIdx + 1, getSourceIndex, 0, 1, layerDimension, layerNode);
 			layerDTOList.add(secondLayerDTO);
 		} else {
-			LayerDTO layerDTO = new LayerDTO(layerId, getSourceIndex, 1, 1);
-			layerDTO.setDimension(layerDimension);
+			LayerDTO layerDTO = prepareLayerDTO(layerIdx, getSourceIndex, 1, 1, layerDimension, layerNode);
 			layerDTOList.add(layerDTO);
 		}
 		// STEP 4 prepare param before call changeContent
 		Map<Object, Object> changeScreenParams = new HashMap<>();
 		changeScreenParams.put(BarcoE2Constant.ID, currentScreenDestId);
-		changeScreenParams.put(BarcoE2Constant.LAYERS, layerDTOList.toArray());
-		JsonNode changeScreenContentResponse = requestByMethod(BarcoE2Constant.METHOD_CHANGE_CONTENT, changeScreenParams);
-		if (changeScreenContentResponse.get(BarcoE2Constant.SUCCESS_STATUS) == null || changeScreenContentResponse.get(BarcoE2Constant.SUCCESS_STATUS).asInt() != 0) {
+		boolean result;
+		if (isMixedLayer) {
+			changeScreenParams.put(BarcoE2Constant.LAYERS, List.of(layerDTOList.get(0)).toArray());
+			boolean assignFirstLayer = requestByMethod(BarcoE2Constant.METHOD_CHANGE_CONTENT, changeScreenParams).get(BarcoE2Constant.SUCCESS_STATUS).asInt() == 0;
+			changeScreenParams.remove(BarcoE2Constant.LAYERS);
+			changeScreenParams.put(BarcoE2Constant.LAYERS, List.of(layerDTOList.get(1)).toArray());
+			boolean assignSecondLayer = requestByMethod(BarcoE2Constant.METHOD_CHANGE_CONTENT, changeScreenParams).get(BarcoE2Constant.SUCCESS_STATUS).asInt() == 0;
+			result = assignFirstLayer && assignSecondLayer;
+		} else {
+			changeScreenParams.put(BarcoE2Constant.LAYERS, layerDTOList.toArray());
+			result = requestByMethod(BarcoE2Constant.METHOD_CHANGE_CONTENT, changeScreenParams).get(BarcoE2Constant.SUCCESS_STATUS).asInt() == 0;
+		}
+		if (!result) {
 			throw new CommandFailureException(this.getAddress(), "changeContent", "Fail to assign source to super/screen destination");
 		}
 	}
 
+	/**
+	 * Prepare data for layer DTO
+	 *
+	 * @param layerId id of layer
+	 * @param getSourceIndex source index
+	 * @param pvmMode preview mode
+	 * @param pgmMode program mode
+	 * @param layerDimension layer dimension
+	 * @param layerNode layer node
+	 * @return LayerDTO
+	 */
+	private LayerDTO prepareLayerDTO(int layerId, int getSourceIndex, int pvmMode, int pgmMode, Dimension layerDimension, JsonNode layerNode) {
+		LayerDTO firstLayerDTO = new LayerDTO(layerId, getSourceIndex, pvmMode, pgmMode);
+		firstLayerDTO.setDimension(layerDimension);
+		int firstLinkLayerId = layerNode.get(layerId).get(BarcoE2Constant.LINK_LAYER_ID).asInt();
```

Comment:
layerNode.get(layerId) can be extracted and reused

Reply:
Fixed